### PR TITLE
[JN-794] Display if a participant has read an email (member's choice)

### PR DESCRIPTION
**Pull request by MatthewBemis**
#### DESCRIPTION

This adds a new icon on the enrollee timeline that shows whether or not a participant has opened an email.

Screenshot 2024-04-01 at 1 27 02 PM

#### TO TEST:  *(simple manual steps for confirming core behavior -- used for pre-release checks)*

* Restart admin API
* Repopulate OurHealth `./scripts/populate_portal.sh ourhealth`
* Go to the participant list and send a bunch of emails https://localhost:3000/ourhealth/studies/ourheart/env/sandbox/participants
* Open a few of them, leave a few others unread
* Wait 30-40 minutes until Sendgrid has propagated the new events and we've captured them in our database (you can speed this up a bit by periodically restarting the admin API, but it will still take a minimum of 10-15 minutes for the events to propagate to sendgrid) 
* Confirm that the email opened statuses in the participant's timeline are correct

### CODE REVIEW

File: ui-admin/src/study/participants/enrolleeView/EnrolleeTimeline.tsx [hunk 1]
```diff
@@ -96,3 +118,18 @@ export default function EnrolleeTimeline({ enrollee, studyEnvContext }:
...
+  const notificationEventDetails = row.eventDetails
+  if (notificationEventDetails && notificationEventDetails.opensCount > 0) {
+    return <FontAwesomeIcon icon={faEnvelopeOpen} aria-label={'Email opened'}/>
+  }
+
+  if (notificationEventDetails) {
+    return <FontAwesomeIcon icon={faEnvelope} aria-label={'Email not yet opened'}/>
+  }
+
+  return <>N/A</>
```
Comment (devonbush) [inline ui-admin/src/study/participants/enrolleeView/EnrolleeTimeline.tsx]: for a softer n/a look
```suggestion
  return <span className="fw-light fst-italic">n/a</span>
```

File: core/src/main/java/bio/terra/pearl/core/model/notification/SendgridEvent.java [hunk 2]
```diff
@@ -1,33 +1,45 @@
...
 import java.time.Instant;
+import java.util.UUID;
 
 @Getter
 @Setter
 @SuperBuilder
 @NoArgsConstructor
 public class SendgridEvent extends BaseEntity {
-    @JsonProperty("msg_id")
+    @JsonAlias("msg_id")
```
Comment (devonbush) [inline core/src/main/java/bio/terra/pearl/core/model/notification/SendgridEvent.java:21]: can you add a comment why we need these json aliases?

File: core/src/main/java/bio/terra/pearl/core/dao/notification/NotificationDao.java [hunk 3]
```diff
@@ -40,6 +44,15 @@ public List<Notification> findByEnrolleeId(UUID enrolleeId) {
         return findAllByProperty("enrollee_id", enrolleeId);
     }
 
+    public void attachSendgridEvent(Notification notification) {
+        Optional<SendgridEvent> sendgridEvent = sendgridEventDao.findByProperty("notification_id", notification.getId());
+        sendgridEvent.ifPresent(notification::setEventDetails);
```
Comment (devonbush) [inline core/src/main/java/bio/terra/pearl/core/dao/notification/NotificationDao.java:49]: ooh, that's some nice syntax I didn't realize you could do

**Review by devonbush — APPROVED** [review]
Looks great -- glad to see this finally getting merged!  Couple minor fluff changes

**Review by connorlbark — APPROVED** [review]

**Pull request merged**